x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumns.java

```diff
@@ -169,7 +174,7 @@ private static void fillInRows(String clusterName, String indexName, Map<String,
             DataType type = field.getDataType();
 
             // skip the nested, object and unsupported types
-            if (isPrimitive(type)) {
+            if (isPrimitive(type) && (isDriver(mode) == false || isTypeSupportedInVersion(type, version))) {
```

Comment:
Why is the drive mode important? What if the non-drivers want to be versioned (as is the plan for the rest APIs)?

Reply:
> Why is the drive mode important?

The drivers would have issues with new types they don't know/understand. And that's why they send their version in each request.

We can relax this check to only consider the version, sure, but IMO we'd then need to understand why other clients would choose to be versioned and what guarantees we'd need to offer. Otherwise an unsigned long or decimal number for a REST client shouldn't pose an issue (since the JSON/CBOR/... library should understand that number).

Reply:
The client check has been now dropped.